src/lib/containers/entity_finder/EntityFinder.tsx

```diff
@@ -142,64 +150,21 @@ export const EntityFinder: React.FunctionComponent<EntityFinderProps> = ({
     [setBreadcrumbsProps],
   )
 
-  /**
-   *
-   * @param entity The entity to check selected status
-   * @param selected the list of selected entities
-   * @returns a boolean array of length equal to entities.length denoting selection status
-   */
-  function isSelected(
-    entity: Reference,
-    selected: Map<string, number>,
-  ): boolean {
-    const match = selected.get(entity.targetId)
-    if (match == null) {
-      return false
-    }
-    if (match === NO_VERSION_NUMBER) {
-      return entity.targetVersionNumber === undefined
-    }
-    return match === entity.targetVersionNumber
-  }
+  const [selectedEntities, toggleSelection] = useEntitySelection(selectMultiple)
 
-  /**
-   * Given the existing selections and a list of toggled references, return the new list of selections
-   * @param selected
-   * @param toggledReference
-   * @returns
-   */
-  function entitySelectionReducer(
-    selected: Map<string, number>,
-    toggledReferences: Reference | Reference[],
-  ): Map<string, number> {
-    const newSelected = selected.withMutations(map => {
-      // Note: we currently don't allow selecting two versions of the same entity, so we replace previous selected version with new selected version
-      if (!Array.isArray(toggledReferences)) {
-        toggledReferences = [toggledReferences]
-      }
-      toggledReferences.forEach(toggledReference => {
-        if (isSelected(toggledReference, selected)) {
-          // remove from selection
-          map.delete(toggledReference.targetId)
-        } else {
-          // add to selection
-          if (!selectMultiple) {
-            map.clear()
-          }
-          map.set(
-            toggledReference.targetId,
-            toggledReference.targetVersionNumber ?? NO_VERSION_NUMBER,
-          )
-        }
-      })
-    })
-
-    return newSelected
-  }
+  const isIdSelected = useCallback(
+    (entity: EntityHeader | ProjectHeader | Hit) => {
+      return selectedEntities.has(entity.id)
+    },
+    [selectedEntities],
+  )
 
-  const [selectedEntities, toggleSelection] = useReducer(
-    entitySelectionReducer,
-    Map<string, number>(),
+  const isSelectable = useCallback(
+    (entity: EntityHeader | ProjectHeader | Hit) => {
+      const type = getEntityTypeFromHeader(entity)
+      return selectableTypes.includes(type)
+    },
+    [selectableTypes],
   )
```

Comment:
Create new functions to determine if a row is selected for select-all purposes. Using `useCallback` to ensure that this function will update when the underlying selected data updates.